src/libcore/str/mod.rs

```diff
@@ -134,12 +134,21 @@ impl FromStr for bool {
 
     /// Parse a `bool` from a string.
     ///
-    /// Yields an `Option<bool>`, because `s` may or may not actually be
-    /// parseable.
+    /// Yields a `Result<bool, ParseBoolError>`, because `s` may or may not
+    /// actually be parseable.
     ///
     /// # Examples
     ///
     /// ```rust
+    /// assert_eq!(from_str::<bool>("true"), OK(true));
+    /// assert_eq!(from_str::<bool>("false"), OK(false));
```

Comment:
I think these instances of `OK` should be `Ok` instead. Also, I believe you can leave off the type annotations due to the usage of `true` and `false` to derive inference (but not in the error example below).

You can be sure to run these doc-blocks via `make check-stage2-doc-crate-core`